src/index.ts

```diff
@@ -0,0 +1,92 @@
+//#region > Imports
+//> Endpoints
+// Contains the apollo endpoint
+import Apollo from "./endpoints/apollo";
+// Contains the scraper endpoint
+import Scraper from "./endpoints/scraper";
+//> Templates
+// Contains the main template
+import { MainTemplate } from "./templates/index";
+//> Sessions
+// Contains the snek and github session
+import { SnekSession, GithubSession } from "./session/sessions";
+//> Interfaces
+// Contains interfaces for scraper and apollo
+import { ScraperEndpoint, ApolloEndpoint } from "./endpoints/index";
+// Contains the interface for the main template
+import { IMainTemplate } from "./templates/index";
+//#endregion
+
+//#region > Interfaces
+/** @interface Endpoint defines the structure of object a endpoint requieres to initialize. */
+interface IEndpoint {
+  url: string;
+  type: string;
+  headers: object;
+}
+/** @interface Client will define the client structure. */
+interface IClient {}
+//#endregion
+
+//#region > Classes
+/**@class The snek-client. Enjoy it. Will be implemented in the future. */
+class Client implements IClient {
+  constructor(ep: IEndpoint) {}
+}
+
+/**@class A client implementation for snek interaction */
+class SnekClient extends Client {
+  endpoint: ApolloEndpoint;
+  template: IMainTemplate;
+  session: SnekSession;
+
+  constructor(
+    type: string = "graphql",
+    url: string = "https://engine.snek.at/api/graphiql",
+    headers: object = {}
```

Comment:
Missing comma.

Reply:
In this case a comma is not needed because this is not an object declaration over multiple lines. This are constructor arguments.